src/H5VLcallback.c

```diff
@@ -3091,6 +3598,51 @@ H5VL_datatype_optional(const H5VL_object_t *vol_obj, H5VL_datatype_optional_t op
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5VL_datatype_optional() */
 
+/*-------------------------------------------------------------------------
```

Comment:
no datatype_optional_vararg? (I see the existing one uses varargs already, but shouldn't they all be made analogous?)

Reply:
Yes, datatype_optional_vararg is here, just a little further down in the file.